Update data analysis human biomolecular page (#190)
elixir-europe/infectious-diseases-toolkit: human-biomolecular-data/data-analysis.md

As data analysis generates new knowledge and information, it is a critical stage in the research process. To follow best practices for analysing human biomolecular data of infectious diseases, it is important to document all steps taken in the analysis workflow, including data preprocessing, statistical analysis, and the use of tools and algorithms. This documentation should be detailed enough to enable others to reproduce the analysis and results. Additionally, it is crucial to follow the FAIR principles (Findability, Accessibility, Interoperability, and Reusability) to ensure that the data, code, and results are easily discoverable, accessible, and reusable by other researchers.

> 

Reviewer: As it is a website, I seem to remember that it was recommended to stick to max 3 sentences for the introduction. Maybe keep the key sentences here and move the other text to a first topic?

When looking for more information [here](https://rdmkit.elixir-europe.org/analysing#what-should-be-considered-for-data-analysis).

> 

Author: done!

### General considerations

Some considerations for analysing human biomolecular data of infectious diseases are:

- Select the tools best suited for the analysis of your data.
- Document the exact steps used for data analysis.
- Choose between several computing infrastructure types, e.g. cluster, cloud.
- Take into account the computing resources needed.
- Which type of data are you using, e.g. DNAseq, ATACseq, CNV.
- Integration of different types of data (e.g. RNAseq and DNAseq).
- Ensure access to the data and tools for all collaborators, for reproducibility.
    - Providing your code 
    - Providing your execution environment
    - Providing your workflows
    - Providing your data analysis execution

> When looking for solutions to some of the considerations above, you can check [this](https://rdmkit.elixir-europe.org/data_analysis#what-are-the-best-practices-for-data-analysis) documentation.

Reviewer: ```suggestion
When looking for solutions to some of the considerations above, you may have a look at the documentation available on the [RDMkit website](https://rdmkit.elixir-europe.org/data_analysis#what-are-the-best-practices-for-data-analysis).
```

### Existing approaches

Here you can find some more general existing approaches in order to help and improve the data analysis pipeline/protocol:

- **Container environments**: As an alternative to package management systems you can consider container environments like [Docker](https://www.docker.com/) or [Singularity](https://sylabs.io/).
- **Web-based platform**: Provides a centralised location for software developers to store, manage, collaborate, and share their code.- You can use [GitHub](https://github.com/) (widely used), [GitLab](https://about.gitlab.com/) or [Bitbucker](https://bitbucket.org/).
- **Workflow platforms**: Allows the user to manage your data and provide an interface (web, GUI, APIs) to run complex pipelines and review their results. For instance: [Galaxy](https://rdmkit.elixir-europe.org/galaxy_assembly) and [Arvados](https://arvados.org/) (CWL-based, open source).

> - **Workflow runners**: Allows the user to take a workflow written in a proprietary or standardiSed format (such as the CWL standard) and execute it locally or on a remote computer infrastructure. For instance, [toil-cwl-runner](https://toil.readthedocs.io/en/latest/running/cwl.html), the reference CWL runner ([cwltool](https://pypi.org/project/cwltool/)), [Nextflow](https://www.nextflow.io/), [Snakemake](https://snakemake.readthedocs.io/en/stable/), [Cromwell](https://cromwell.readthedocs.io/en/stable/tutorials/FiveMinuteIntro/).

Author: This change has been addressed

- **Integration Pipelines**: These pipelines are used to integrate different types of data, such as genomic, proteomic, and metabolomic data. It involves steps such as data preprocessing, data integration, and functional analysis. Tools like [OmicsIntegrator](https://github.com/fraenkel-lab/OmicsIntegrator) can be used for data integration.

## Preprocessing

Data preprocessing is the phase in the project where data is converted into a desired format and prepared for analysis. Is a crucial step in data analysis that involves cleaning, transforming, and preparing data for analysis. The goal of preprocessing is to ensure that the data is of high quality and is suitable for the intended analysis. Preprocessing can involve a range of steps, depending on the type of data and the analysis being performed. 

> Preprocessing is a critical step in data analysis of HUman biomolecular data of infectious diseases because it can greatly impact the accuracy and reliability of the analysis results. By ensuring that the data is of high quality and suitable for analysis, preprocessing can help researchers obtain more accurate and meaningful insights from this data.

Reviewer: ```suggestion
Preprocessing is a critical step in data analysis of Human biomolecular data of infectious diseases because it can greatly impact the accuracy and reliability of the analysis results. By ensuring that the data is of high quality and suitable for analysis, preprocessing can help researchers obtain more accurate and meaningful insights from this data.
```

Author: This change has been addressed

- **Remove low-quality samples**: Samples that have low sequencing depth, high number of missing values, or poor alignment quality can be removed from the dataset to ensure that the remaining samples are of high quality.
- **Identify and remove outliers**: Outliers are data points that fall outside the expected range of values and can skew the analysis results. Outliers can be identified using statistical methods and removed from the dataset.
- **Check for batch effects**: Batch effects are systematic differences in the data that arise from technical or experimental factors. Batch effects can be identified using statistical methods and removed from the dataset.
- **Data normalisation**: Normalisation is a common preprocessing step that aims to remove systematic biases in the data. Normalisation methods should be evaluated to ensure that they are effective and do not introduce additional biases.
- **Perform quality control checks at each preprocessing step**: Quality control checks should be performed at each preprocessing step to ensure that the data is of high quality and suitable for the intended analysis. 

### Existing approaches

> Preprocessing could be made using the state of art bioinformatics tools and/or programming languages that have different functions and packages to work and process this kind of data. For example, Python, RStudio or using Command-Line, are different approaches to enable the user performing all the necessary and wanted steps to do the desired preprocessing pipeline/protocol. 

Reviewer: ```suggestion
Preprocessing could be done using the state of art bioinformatics tools and/or programming languages that have different functions and packages to work and process this kind of data. For example, Python, RStudio or using Command-Line, are different approaches to enable the user performing all the necessary and wanted steps to do the desired preprocessing pipeline/protocol. 
```

When looking for quality control protocols, see [Human biomolecular data - Quality control](https://www.infectious-diseases-toolkit.org/human-biomolecular-data/quality-control) page.

## Analysis

The analysis of human biomolecular data involves the use of various techniques and approaches to extract meaningful information from biological samples such as DNA, RNA, proteins, and metabolites. 

> 

Reviewer: Are there any specific aspects to consider when analysing human biomolecular data? (just to be consistent structure wise)

- **Protein-protein interaction analysis**: This involves identifying proteins that interact with each other and exploring the functional consequences of these interactions. This can help to identify new targets for drug development and understand disease mechanisms.
    - *Interaction databases*: [BioGRID](https://github.com/BioGRID) and [IntAct](https://github.com/intact-portal)
    - *Network analysis*: [Cytoscape](https://apps.cytoscape.org/) and [GeneMANIA](https://github.com/GeneMANIA/genemania)
- **Metabolomics analysis**: This involves measuring the levels of small molecules (metabolites) in biological samples and comparing them across different conditions or groups of samples. This can help to identify biomarkers of disease or drug response.
    - *Data processing*: [XCMS](https://xcmsonline.scripps.edu/landing_page.php?pgcontent=mainPage), [MZmine](http://mzmine.github.io/) and [OpenMS](https://github.com/OpenMS/OpenMS)
    - *Statistical analysis*: [MetaboAnalyst](https://www.metaboanalyst.ca/) and [MetSign](https://pubmed.ncbi.nlm.nih.gov/21932828/)

## Postprocessing

> The postprocessing part refers to the steps taken after the initial analysis to refine and interpret the results. Post-processing steps are important because they can help to identify biological patterns and relationships that were not apparent in the initial analysis, and to ensure that the results are biologically meaningful and reproducible.

Reviewer: Tiny detail, in text here says 'Post-processing' while title says 'Postprocessing' :)

Author: This change has been addressed
